crates/synthesis/src/linalg/mod.rs

```diff
@@ -10,4 +10,127 @@
 // copyright notice, and modified files need to carry a notice indicating
 // that they have been altered from the originals.
 
+use approx::{abs_diff_eq, relative_ne};
+use faer::MatRef;
+use faer_ext::{IntoFaer, IntoNalgebra};
+use nalgebra::{DMatrix, DMatrixView};
+use num_complex::Complex64;
+
 pub mod cos_sin_decomp;
+
+const ATOL_DEFAULT: f64 = 1e-8;
+const RTOL_DEFAULT: f64 = 1e-5;
+
+/// Check whether the given matrix is hermitian by comparing it (up to tolerance) with its hermitian adjoint.
+pub fn is_hermitian_matrix(mat: &DMatrix<Complex64>) -> bool {
```

Comment:
I don't know if it matters that much, but you can pass `DMatrixView` instead of `&DMatrix`. I think the difference is who owns the underlying memory, and using `DMatrixView` makes the least assumptions.